### PR TITLE
enable the conversation of single files to typescript

### PR DESCRIPTION
**Pull request by ksjogo**
#### Checklist

* [x] Use a separate branch in your local repo (not `master`).
* [x] Test coverage is 100% (or you have a story for why it's ok).

### CODE REVIEW

**Comment by ksjogo** [comment]
@googlebot I signed it!

**Comment by nicolodavis** [comment]
Please rebase this over the `phases-overhaul` branch.  Otherwise, it will be hard to merge this with all the changes in there.

**Comment by ksjogo** [comment]
Did so for now and added some initial typing based on @Oliboy50 progress linked in #190.
Will have to check with the actual phase changes though.
And some tests fail atm, will get to that after sleep.

File: src/core/game.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+}
+
+export interface GameFlowPhase<GR, G = ExtractG<GR>, M = ExtractM<GR>> {
+  endPhaseIf?: (G: G, ctx: Context) => boolean | { next: PhaseID };
+  next?: PhaseID;
+  endTurnIf?: (G: G, ctx: Context) => boolean | { next: PlayerID };
+  onTurnBegin?: (G: G, ctx: Context) => G | void;
+  onTurnEnd?: (G: G, ctx: Context) => G | void;
+  onMove?: (G: G, ctx: Context) => G | void;
+  onPhaseBegin?: (G: G, ctx: Context) => G | void;
```
Comment (nicolodavis) [inline src/core/game.ts]: These functions have been renamed in this branch and will need to be updated.

File: packages/ai-visualize.js [hunk 2]
```diff
@@ -6,6 +6,6 @@
  * https://opensource.org/licenses/MIT.
  */
 
-import { MCTSVisualizer } from '../src/ai/mcts-visualizer.js';
```
Comment (nicolodavis) [inline packages/ai-visualize.js]: I would suggest that we split this PR into two:

1. One that removes the `.js` suffices.
2. One that adds the typings.

**Comment by ksjogo** [comment]
Removed the initial typings for now, but left files and tests as tsx, so we would see if that causes any errors in publishing/bundling.

File: python/examples/tic-tac-toe/boardgameio.py [hunk 3]
```diff
@@ -1 +0,0 @@
-../../boardgameio.py
```
Comment (ksjogo) [inline python/examples/tic-tac-toe/boardgameio.py]: I am not sure what happened here.

**Comment by nicolodavis** [comment]
The tsx files are causing some test failures.  Can you take a look?

Let's just use this PR to strip the prefixes of the `.js` imports?

**Comment by nicolodavis** [comment]
@ksjogo Are you still interested in getting this in?  The `phases-overhaul` branch has been merged into `master` so you can just work off the `master` branch.

**Comment by ksjogo** [comment]
Hey @nicolodavis ,yeah I am still interested in this. Just moved countries, will fix/rebase in the next days.

**Pull request closed without merging**